src/core/unit.c

```diff
@@ -880,12 +880,18 @@ UnitActiveState unit_active_state(Unit *u) {
          * loaded but still has a process around. That's why we won't
          * shortcut failed loading to UNIT_INACTIVE_FAILED. */
 
+        if (!UNIT_VTABLE(u)->active_state)
```

Comment:
Hmm, any unit type without this is completely broken. Please just use ASSERT_PTR(...->active_state)(u)

Reply:
It can happen that units don't have this initialized, but other fields are. All other ptrs are checked already, there's no reason not to check this too